src/mds/CDir.cc

```diff
@@ -1801,14 +1808,20 @@ CDentry *CDir::_load_dentry(
         dout(12) << "_fetched  got remote link " << ino << " (don't have it)" << dendl;
       }
     }
-  } 
-  else if (type == 'I') {
+  }
+  else if (type == 'I' || type == 'i') {
     // inode
-    
+
+    if (type == 'i') {
+      decode(dn->dentry_flags, q);
+      if (unlikely(dn->has_alternate_name()))
+        decode(dn->alternate_name, q);
+    }
```

Comment:
when type is 'i'. this code should looks like

DECODE_START(1, q);
inode_data.decode(q);
decode(dn->alternate_name, q);
DECODE_FINISH(q);

Reply:
Done, thanks.